runtime/src/stakes.rs

```diff
@@ -183,7 +219,8 @@ impl Stakes {
         let stake_history_entry = thread_pool.install(|| {
             stake_delegations
                 .par_iter()
-                .fold(StakeActivationStatus::default, |acc, delegation| {
+                .fold(StakeActivationStatus::default, |acc, stake_account| {
+                    let delegation = stake_account.delegation().unwrap();
```

Comment:
Removed `unwrap`s here and elsewhere.
The type is now `StakeAccount<Delegation>` which ensures the stake-state is always a `Delegation` and so `StakeAccount::<Delegation>::delegation()` returns `Delegation` instead of `Option<Delegation>`.

Reply:
even better!